storage/imx6ull-flash/flashdev.c

```diff
@@ -99,113 +102,141 @@ static unsigned int _flashmtd_checkErased(const void *buff, size_t boffs, size_t
 }
 
 
-/* Returns maximum number of bitlips or number smaller than 0 in case of error */
-static int _flashmtd_getMaxflips(struct _storage_t *strg, uint32_t paddr, int nchunks)
+static void _flashmtd_chunkCorrect(struct _storage_t *strg, unsigned int chunkidx, unsigned int nflips)
 {
-	/* Metadata and data (with their ECC) chunk sizes in bits */
-	const size_t mlen = strg->dev->mtd->oobSize * CHAR_BIT + ECC_GF * ECC0_BITFLIP_STRENGHT;
-	const size_t dlen = ECCN_DATA_SIZE * CHAR_BIT + ECC_GF * ECCN_BITFLIP_STRENGHT;
 	flashdrv_meta_t *meta = strg->dev->ctx->metabuf;
 	unsigned char *data = strg->dev->ctx->databuf;
-	unsigned int i, flips, maxflips = 0;
-	size_t boffs, blen, rawsz;
+
+	if (nflips == 0) {
+		/* Nothing to be corrected */
+		return;
+	}
+
+	if (CHUNK_IS_UNCORECTABLE(chunkidx, nflips)) {
```

Comment:
IMHO it would be more readable to move this condition outside of this function (`_flashmtd_chunkCorrect` should correct the chunk, if it's conditional should be renamed to eg. `_flashmtd_chunkTryCorrect`)

Reply:
Yes, I agree it would be more readable to move this check outside this function.